packages/trpc/server/routers/viewer/slots.tsx

```diff
@@ -280,85 +279,93 @@ export async function getSchedule(input: z.infer<typeof getScheduleSchema>, ctx:
       periodDays: eventType.periodDays,
     });
 
-  let currentCheckedTime = startTime;
-  let getSlotsTime = 0;
+  const getSlotsTime = 0;
   let checkForAvailabilityTime = 0;
-  let getSlotsCount = 0;
+  const getSlotsCount = 0;
   let checkForAvailabilityCount = 0;
-  do {
-    const startGetSlots = performance.now();
-    // get slots retrieves the available times for a given day
-    const timeSlots = getTimeSlots({
-      inviteeDate: currentCheckedTime,
-      eventLength: input.duration || eventType.length,
-      workingHours,
-      dateOverrides,
-      minimumBookingNotice: eventType.minimumBookingNotice,
-      frequency: eventType.slotInterval || input.duration || eventType.length,
-    });
 
-    const endGetSlots = performance.now();
-    getSlotsTime += endGetSlots - startGetSlots;
-    getSlotsCount++;
-
-    const isCollective = !eventType.schedulingType || eventType.schedulingType === SchedulingType.COLLECTIVE;
-
-    const availableTimeSlots = timeSlots
-      .filter((slot) => isTimeWithinBounds(slot.time))
-      .filter((slot) =>
-        isCollective
-          ? // The slot should be available for every user
-            userAvailability.every((schedule) => {
-              const startCheckForAvailability = performance.now();
-              const isAvailable = checkIfIsAvailable({
-                time: slot.time,
-                ...schedule,
-                ...availabilityCheckProps,
-              });
-              const endCheckForAvailability = performance.now();
-              checkForAvailabilityCount++;
-              checkForAvailabilityTime += endCheckForAvailability - startCheckForAvailability;
-              return isAvailable;
-            })
-          : (() => {
-              // The slot should be available for the atleast one of the slot owners.
-              return slot.userIds?.some((slotUserId) => {
-                const userSchedule = userAvailability.find(({ userId }) => userId === slotUserId);
-                if (!userSchedule) {
-                  throw new TRPCError({
-                    message: "Shouldn't happen that we don't have a matching user schedule here",
-                    code: "INTERNAL_SERVER_ERROR",
-                  });
-                }
-                return checkIfIsAvailable({
-                  time: slot.time,
-                  ...userSchedule,
-                  ...availabilityCheckProps,
-                });
-              });
-            })()
-      );
+  const timeSlots: ReturnType<typeof getTimeSlots> = [];
+
+  for (
```

Comment:
`do {} while {}` changed in a `for` loop, flattened into one big array for decreasing depth.